frontend/src/mixin/dynamicTooltipMixin.ts

```diff
@@ -3,17 +3,26 @@ import { defineComponent } from 'vue';
 export default defineComponent({
   methods: {
     onTooltipHover(refName: string): void {
-      const tooltipTextElement = (this.$refs[refName] as HTMLElement[])[0];
+      const tooltipTextElement = this.getElementByRef(refName);
       if (this.isElementAboveViewport(tooltipTextElement)) {
         tooltipTextElement.classList.add('bottom-aligned');
       }
     },
     resetTooltip(refName: string): void {
-      const tooltipTextElement = (this.$refs[refName] as HTMLElement[])[0];
+      const tooltipTextElement = this.getElementByRef(refName);
       tooltipTextElement.classList.remove('bottom-aligned');
     },
     isElementAboveViewport(el: Element): boolean {
       return el.getBoundingClientRect().top <= 0;
     },
+    /**
+     * Note: this.$refs[refName] can be an array of HTMLElements
+     * if the ref is on a v-for loop, else it will be a single HTMLElement.
+     */
+    getElementByRef(refName: string): HTMLElement {
+      return Array.isArray((this.$refs[refName]))
```

Comment:
Minor nit, but there brackets here are doubled

Reply:
Thanks for catching that! I fixed it.